package.json

```diff
@@ -62,12 +62,14 @@
 		"build-css:editor": "check-npm-client && node-sass --importer node_modules/node-sass-package-importer/dist/cli.js --include-path client client/assets/stylesheets/editor.scss public/editor.css --output-style compressed && yarn run -s postcss public/editor.css",
 		"build-css:reader-mobile": "check-npm-client && node-sass --importer node_modules/node-sass-package-importer/dist/cli.js --include-path client client/assets/stylesheets/reader-mobile.scss public/reader-mobile.css --output-style compressed && yarn run -s postcss public/reader-mobile.css",
 		"build-devdocs:search-index": "check-npm-client && node bin/generate-devdocs-search-index.js",
-		"build-desktop": "check-npm-client && run-s build-desktop:config build-desktop:server build-desktop:client build-desktop:install-app-deps build-desktop:app",
+		"build-desktop": "check-npm-client && yarn run build-desktop:source && run-s build-desktop:install-app-deps build-desktop:app",
+		"build-desktop:source": "check-npm-client && run-s build-desktop:config build-desktop:server build-desktop:client",
 		"build-desktop:config": "check-npm-client && node desktop/bin/build-desktop-config.js",
 		"build-desktop:server": "check-npm-client && BROWSERSLIST_ENV=server CALYPSO_ENV=desktop webpack --config client/webpack.config.desktop.js --display errors-only",
 		"build-desktop:client": "check-npm-client && CALYPSO_ENV=desktop DESKTOP_MONOREPO=true yarn run build-client-fallback",
-		"build-desktop:install-app-deps": "check-npm-client && cd desktop && yarn install --frozen-lockfile",
-		"build-desktop:app": "check-npm-client && cd desktop && yarn run build",
+		"build-desktop:install-app-deps": "cd desktop && yarn install --frozen-lockfile",
+		"build-desktop:app": "cd desktop && yarn run build",
```

Comment:
Did the `check-npm-client` cause any mischief here? We probably don't need to run the check all the time, especially in sub-tasks that are not meant to run directly from the CLI, but I'm just curious why you removed it.

Reply:
> Did the check-npm-client cause any mischief here?

The `build-desktop:app` script needs to be executed within a platform-specific (Windows/Mac/Linux) context where Calypso's modules are not installed. `check-npm-client` is not one of the desktop's dependencies, and so this script would fail. I think this is fine, however - the `build-desktop:source` script is prefixed with the `check-npm-client`, so we know this check has occurred upstream by the time the Electron packaging step occurs.